tests/integration/ceph_smoke_test.go

```diff
@@ -341,3 +343,106 @@ func (s *SmokeSuite) getNonCanaryMonDeployments() ([]appsv1.Deployment, error) {
 	}
 	return nonCanaryMonDeployments, nil
 }
+
+// Smoke Test for Webhooks
+func (s *SmokeSuite) TestWebhooks() {
+	logger.Info("Create webhook Smoke Test")
+	ctx := context.TODO()
+	t := s.T()
+
+	t.Run("Fail updating cephCluster, hostPath cannot be changes", func(t *testing.T) {
+		cephCRList, err := s.k8sh.RookClientset.CephV1().CephClusters(s.settings.Namespace).List(ctx, metav1.ListOptions{})
+		require.NoError(s.T(), err)
+		cephCR := cephCRList.Items[0]
+		cephCR.Spec.DataDirHostPath = "/var"
+
+		_, err = s.k8sh.RookClientset.CephV1().CephClusters(s.settings.Namespace).Update(ctx, &cephCR, metav1.UpdateOptions{})
+		require.Error(s.T(), err)
+	})
+
+	t.Run("Fail pool creation, both erasurecoded and replicated fields cannot be set at the same time", func(t *testing.T) {
+		pool := &cephv1.CephBlockPool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "webhook-pool",
+				Namespace: s.settings.Namespace,
+				UID:       types.UID("c47cac40-9bee-4d52-823b-ccd803ba5bfe"),
+			},
+			Spec: cephv1.NamedBlockPoolSpec{
+				PoolSpec: cephv1.PoolSpec{
+					Replicated: cephv1.ReplicatedSpec{
+						Size: uint(1),
+					},
+					ErasureCoded: cephv1.ErasureCodedSpec{
+						CodingChunks: 1,
+						DataChunks:   1,
+					},
+				},
+			},
+		}
+		_, err := s.k8sh.RookClientset.CephV1().CephBlockPools(s.settings.Namespace).Create(ctx, pool, metav1.CreateOptions{})
+		require.Error(s.T(), err)
+	})
+
+	t.Run("Fail objectStore creation, either of port or securePort fields should be not be zero ", func(t *testing.T) {
+		objectStore := &cephv1.CephObjectStore{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "webhook-store",
+				Namespace: s.settings.Namespace,
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind: "CephObjectStore",
+			},
+			Spec: cephv1.ObjectStoreSpec{
+				Gateway: cephv1.GatewaySpec{
+					Port:       0,
+					SecurePort: 0,
+				},
+			},
+		}
+		_, err := s.k8sh.RookClientset.CephV1().CephObjectStores(s.settings.Namespace).Create(ctx, objectStore, metav1.CreateOptions{})
+		require.Error(s.T(), err)
+	})
+
+	subVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "subvol-name",
+		},
+		Spec: cephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: "filesystem-name",
+		},
+	}
+	subVolumeGroup, err := s.k8sh.RookClientset.CephV1().CephFilesystemSubVolumeGroups(s.settings.Namespace).Create(ctx, subVolumeGroup, metav1.CreateOptions{})
```

Comment:
I think it's a good pattern that if `t.Run()` is used, everything should be included in a `t.Run()` method besides perhaps structs that are used in multiple tests. So you could either create a new `t.Run()` or include it in a previous one for the cleanup.

Reply:
included similar tests in one `t.Run()`